build-tools/packages/build-cli/src/commands/typetests.ts

```diff
@@ -153,29 +158,62 @@ export function previousVersion(version: string): string {
 
 /**
  * Updates configuration for type tests in package.json
+ *
+ * @internal
  */
-function updateTypeTestConfiguration(pkgJson: PackageJson, options: TypeTestConfigActions): void {
+export function updateTypeTestConfiguration(
+	pkgJson: PackageJson,
+	options: TypeTestConfigActions,
+): void {
 	if (options.version !== undefined) {
-		const oldDepName = `${pkgJson.name}-previous`;
+		applyTypeTestVersionOptions(pkgJson, options.version);
+	}
 
-		// Packages can explicitly opt out of type tests by setting typeValidation.disabled to true.
-		const enabled = pkgJson.typeValidation?.disabled !== true;
+	if (options.resetBroken !== undefined) {
+		resetBrokenTests(pkgJson, options.resetBroken);
+	}
+}
 
-		if (!enabled || options.version === VersionOptions.Clear) {
-			// There isn't a type safe alternative to delete here,
-			// and delete should be safe since the key can't collide with anything builtin since it has to end in `-previous` which makes it not a valid identifier.
+/**
+ * Adds or removes the devDependency on a previous version of a package thatis used for type testing. This function only
+ * affects the `devDependencies` nodes in package.json.
+ *
+ * @internal
+ */
+export function applyTypeTestVersionOptions(
+	pkgJson: PackageJson,
+	versionOptions: string | VersionOptions,
+): void {
+	const oldDepName = `${pkgJson.name}-previous`;
+
+	// Packages can explicitly opt out of type tests by setting typeValidation.disabled to true.
+	const enabled = pkgJson.typeValidation?.disabled !== true;
+
+	if (!enabled || versionOptions === VersionOptions.Clear) {
+		// There isn't a type safe alternative to delete here,
+		// and delete should be safe since the key can't collide with anything builtin since it has to end in `-previous`
+		// which makes it not a valid identifier.
+		if (pkgJson.devDependencies?.[oldDepName] !== undefined) {
 			// eslint-disable-next-line @typescript-eslint/no-dynamic-delete
 			delete pkgJson.devDependencies[oldDepName];
-		} else if (options.version !== VersionOptions.ClearIfDisabled) {
-			const newVersion: string =
-				options.version === VersionOptions.Previous
-					? previousVersion(pkgJson.version)
-					: options.version;
-			pkgJson.devDependencies[oldDepName] = `npm:${pkgJson.name}@${newVersion}`;
 		}
+	} else if (versionOptions !== VersionOptions.ClearIfDisabled) {
+		const newVersion: string =
+			versionOptions === VersionOptions.Previous
+				? previousVersion(pkgJson.version)
+				: versionOptions;
+		pkgJson.devDependencies ??= {};
+		pkgJson.devDependencies[oldDepName] = `npm:${pkgJson.name}@${newVersion}`;
 	}
+}
 
-	if (options.resetBroken === true && pkgJson.typeValidation !== undefined) {
+/**
+ * Removes any `typeValidation.broken` entries from package.json.
+ *
+ * @internal
```

Comment:
You shouldn't mark package internal APIs with @internal. No release set at all means package internal, and @internal means public to other packages with the same maintainer: https://api-extractor.com/pages/tsdoc/tag_internal/